test/parallel/test-dotenv-node-options.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+require('../common');
+const assert = require('node:assert');
+const { spawnSync } = require('node:child_process');
+
+const envFilePath = '../fixtures/dotenv/node-options.env';
+
+{
+  // Make sure we have access to permission scope.
+  const code = '"require(\'assert\').assert(process.permission.has(\'fs.read\'))\'"';
+  const child = spawnSync(process.execPath,
+                          [ `--env-file=${envFilePath}`, '-e', code ], {
+                            cwd: __dirname
+                          });
+  // Test NODE_NO_WARNINGS environment variable
+  // `stderr` should not contain "ExperimentalWarning: Permission is an experimental feature" message
+  assert.strictEqual(child.stderr.toString(), '');
+  assert.strictEqual(child.status, 0);
+}
+
+{
+  // Make sure only the read permissions are enabled.
+  const code = `
+    "require('fs').writeFileSync(require('path').join(__dirname, 'should-not-write.txt'), 'hello', 'utf-8')"
+  `;
+  const child = spawnSync(process.execPath,
+                          [ `--env-file=${envFilePath}`, '-e', code.trim() ], {
+                            cwd: __dirname,
+                            encoding: 'utf-8',
+                          });
+  assert.strictEqual(child.status, 0);
+}
+
+{
+  // Test TZ environment variable
+  const code = `
+    "require('assert').assert(new Date().toString().includes('Irish Standard Time'))"
```

Comment:
This test will fail in the winter, when Dublin uses Greenwich Mean Time. Maybe instead of Dublin set `TZ` to `Pacific/Honolulu`, and search for `Hawaii` in the resulting string? The full timezone is “Hawaii-Aleutian Standard Time”; Hawaii doesn’t observe daylight savings.